Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsitePolicies.mm

```diff
@@ -1833,3 +1834,146 @@ - (void)webView:(WKWebView *)webView runJavaScriptAlertPanelWithMessage:(NSStrin
+
+    TestWebKitAPI::HTTPServer server({
+        { "/subframe.html"_s, { "<script src='test:///script_subframe.js'></script>"_s } },
+    }, TestWebKitAPI::HTTPServer::Protocol::Http);
+
+    __block auto port = server.port();
+    auto handler = adoptNS([TestURLSchemeHandler new]);
+    [handler setStartURLSchemeTaskHandler:^(WKWebView *, id<WKURLSchemeTask> task) {
+        NSString *path = task.request.URL.path;
+        NSString *type = nil;
+        NSString *result = nil;
+        if ([path hasSuffix:@"index.html"]) {
+            result = [NSString stringWithFormat:@"<!DOCTYPE html>"
+                "<script>"
+                "   window.results = [];"
+                "   window.addEventListener('message', function(event) {"
+                "       window.results.push(event.data);"
+                "   });"
+                "</script>"
+                "<script src='test:///script.js'></script>"
+                "<iframe src='http://localhost:%d/subframe.html'></iframe>", port];
+            type = @"text/html";
+        } else if ([path hasSuffix:@"script.js"]) {
+            result = @"window.results.push(window.location.href);";
+            type = @"text/javascript";
+        } else if ([path hasSuffix:@"script_subframe.js"]) {
+            result = @"window.parent.postMessage(window.location.href, '*');";
+            type = @"text/javascript";
+        }
+
+        if (!result) {
+            [task didFailWithError:[NSError errorWithDomain:@"TestWebKitAPI" code:1 userInfo:nil]];
+            return;
+        }
+
+        auto response = adoptNS([[NSURLResponse alloc] initWithURL:task.request.URL MIMEType:type expectedContentLength:[result length] textEncodingName:nil]);
+        [task didReceiveResponse:response.get()];
+        [task didReceiveData:[result dataUsingEncoding:NSUTF8StringEncoding]];
+        [task didFinish];
+    }];
+
+    auto configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
+    [configuration setURLSchemeHandler:handler.get() forURLScheme:@"test"];
+
+    doneCompiling = false;
+    NSString* contentBlocker = @"[{\"action\":{\"type\":\"block\"},\"trigger\":{\"url-filter\":\".*\",\"resource-type\":[\"script\"]}}]";
+    [[WKContentRuleListStore defaultStore] compileContentRuleListForIdentifier:@"WebsitePoliciesTest" encodedContentRuleList:contentBlocker completionHandler:^(WKContentRuleList *list, NSError *error) {
+        EXPECT_TRUE(error == nil);
+        [[configuration userContentController] addContentRuleList:list];
+        doneCompiling = true;
+    }];
+    TestWebKitAPI::Util::run(&doneCompiling);
+
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 800, 600) configuration:configuration.get()]);
+    [webView setNavigationDelegate:navigationDelegate.get()];
+
+    auto request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:@"test:///index.html"]];
+    [webView loadRequest:request];
+    [navigationDelegate waitForDidFinishNavigation];
```

Comment:
This doesn't wait for the postMessage to go through, so this test will likely be flaky.  Using something like _test_waitForAlert and alerting the results instead will fix this.

Reply:
The postMessage should never fire in this case, since the script should be blocked. Instead, I added an additional 0.5 second wait. The other tests should be able to do this though, so I changed those to wait for an alert.